src/MSGEQ7.hpp

```diff
@@ -220,3 +215,22 @@ getVolume(void){
 	// return the average of all channels
 	return vol / 7;
 }
+
+template <uint16_t smooth, uint8_t resetPin, uint8_t strobePin, uint8_t firstAnalogPin, uint8_t ...analogPins>
+MSGEQ7_data_t CMSGEQ7<smooth, resetPin, strobePin, firstAnalogPin, analogPins ...>::
+getAnalogReadPin(uint8_t pin) {
+	// This assumes the analogReadResolution is already at 10 bit.
+	int read = analogRead(pin);
+	#if defined(MSGEQ7_10BIT) || defined(MSGEQ7_12BIT)
+		// Return the value as int.
+		return read;
+	#else
+		// get max resolution of analog pin and map it to 8 bit range. Some systems, esp8266, have range from 0 - 1024 so there is a max range of 1023.
```

Comment:
Sorry, I am a bit confused. Doesnt the esp has 12 bite analog value? why are you writing about an 10 bit value here? Also the bitshifting depends on the input value. If the input is 10 bit (avr) you need to shift twice, on esp 12 bit you need to shift 4 times. wouldnt it be better to detect the MCU family and shift the bits accordingly?

Reply:
> Sorry, I am a bit confused. Doesnt the esp has 12 bite analog value?


No. Well, depends on the board. On the ESP8266, which is a variant of ESP, only allows 10 Bit analog value, but on the ESP32, which is another variant of ESP, is able to change the analog value from 10 Bit to 12 Bit, just like the Due, Zero and MKR Family. More info here: https://www.arduino.cc/reference/en/language/functions/zero-due-mkr-family/analogreadresolution/.



> why are you writing about an 10 bit value here?

I wrote the 10 bit value there because the user is able to change the analog values. Let's say I decided to change the analog value from 10 bit to 12 Bit, well then that mean that you will have to shift the bits 4 times, which is what you said, but there is no way to see what analog value is the pin, but I could not find a method that let us know what the analog value is for the current pin.

> wouldnt it be better to detect the MCU family and shift the bits accordingly?

That would be better but Is there a method to detect the current resolution of a bit? Unless if we set the analogReadResolution?

Reply:
You can do arduino board checks like #ifdef ARDUINO_NANO (or similar). We could add that for esp and do a 4x shift? That is hard to explain...

AVR board always have a 10 bit measure resolution. We decide on the output resolution in our code, and we normally want 8 bits. So we need to shift twice `>>2`

On ESP (as far as I understand) the analog value is always measured in 12 bits (correct?). we need to shift twice for 10 bit and 4th for 8 bit then. So we need to shift, depending on the analog input range.

Reply:
> On ESP (as far as I understand) the analog value is always measured in 12 bits (correct?). we need to shift twice for 10 bit and 4th for 8 bit then. So we need to shift, depending on the analog input range.

Some ESP boards, like the ESP8266, measures the analog values ONLY at 10 bit. Others, like the ESP32, allows to change the bit value from 8 - 12 bit (and theoretically even more) but it's recommended to stay in the 8-12 bit. 

> we need to shift twice for 10 bit and 4th for 8 bit then. So we need to shift, depending on the analog input range.

I'm going to cancel this pull request since this is trying to add support to 12 bit, but I do not think It's ready yet, but what I am trying to fix is the issue with the ESP board value changing to zero and the primary reason for that is for some reason the 10 bit analog value of the ESP8266 ranges from 0 - 1024 rather than 0 - 1023. If you shift 1024 twice, you get 256, but 256 does not fit in the `uint8_t` range, which returns 0.